tests/analysis_tests.py

```diff
@@ -45,6 +45,17 @@ def test_segment_rowsize_raises(self):
         with self.assertRaises(ValueError):
             segment(x, tol, rowsize)
 
+    def test_segments_datetime(self):
+        x = [
+            np.datetime64("2023-01-01"),
```

Comment:
the function uses `datetime.timedelta` but here you use `np.datetime64`. Does it matter?

Reply:
I modified to datetime for consistency, but looks like both are compatible.